osu.Game/Rulesets/Edit/HitObjectInspector.cs

```diff
@@ -0,0 +1,140 @@
+        {
+            AutoSizeAxes = Axes.Y;
+            RelativeSizeAxes = Axes.X;
+
+            InternalChild = inspectorText = new OsuTextFlowContainer
+            {
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            EditorBeatmap.SelectedHitObjects.CollectionChanged += (_, _) => updateInspectorText();
+            EditorBeatmap.TransactionBegan += updateInspectorText;
+            EditorBeatmap.TransactionEnded += updateInspectorText;
+        }
+
+        private void updateInspectorText()
+        {
+            inspectorText.Clear();
+
+            switch (EditorBeatmap.SelectedHitObjects.Count)
+            {
+                case 0:
+                    addValue("No selection");
+                    break;
+
+                case 1:
+                    var selected = EditorBeatmap.SelectedHitObjects.Single();
+
+                    addHeader("Type");
+                    addValue($"{selected.GetType().ReadableName()}");
+
+                    addHeader("Time");
+                    addValue($"{selected.StartTime:#,0.##}ms");
+
+                    switch (selected)
+                    {
+                        case IHasPosition pos:
+                            addHeader("Position");
+                            addValue($"x:{pos.X:#,0.##} y:{pos.Y:#,0.##}");
+                            break;
+
+                        case IHasXPosition x:
+                            addHeader("Position");
+
+                            addValue($"x:{x.X:#,0.##} ");
+                            break;
+
+                        case IHasYPosition y:
+                            addHeader("Position");
+
+                            addValue($"y:{y.Y:#,0.##}");
+                            break;
+                    }
+
+                    if (selected is IHasDistance distance)
+                    {
+                        addHeader("Distance");
+                        addValue($"{distance.Distance:#,0.##}px");
+                    }
+
+                    if (selected is IHasRepeats repeats)
+                    {
+                        addHeader("Repeats");
+                        addValue($"{repeats.RepeatCount:#,0.##}");
+                    }
+
+                    if (selected is IHasDuration duration)
+                    {
+                        addHeader("End Time");
+                        addValue($"{duration.EndTime:#,0.##}ms");
+                        addHeader("Duration");
+                        addValue($"{duration.Duration:#,0.##}ms");
+                    }
+
+                    // I'd hope there's a better way to do this, but I don't want to bind to each and every property above to watch for changes.
+                    // This is a good middle-ground for the time being.
+                    Scheduler.AddDelayed(updateInspectorText, 250);
```

Comment:
I would really love to look away from this, I really would, but I cannot. And that is because there's a _slight_ issue here that can decimate performance in editor because of this (note pending task count):

![1680714900](https://user-images.githubusercontent.com/20418176/230155301-c5a4cf0b-8e07-4e68-b3d4-a392f91f0061.png)

To reproduce, select any single object and just start spamming the sample change hotkeys. Every sample change is a `TransactionBegan` call followed by a `TransactionEnded` - so two `updateInspectorText()` calls. Due to this delayed-recursive call, every `updateInspectorText()` is essentially repeating until the selection changes to either nothing or multiple objects, leading to this runaway scheduled delegate count scenario.

The easiest fix would probably be this:

```diff
diff --git a/osu.Game/Rulesets/Edit/HitObjectInspector.cs b/osu.Game/Rulesets/Edit/HitObjectInspector.cs
index 1870476ca0..02270d4662 100644
--- a/osu.Game/Rulesets/Edit/HitObjectInspector.cs
+++ b/osu.Game/Rulesets/Edit/HitObjectInspector.cs
@@ -6,6 +6,7 @@
 using osu.Framework.Extensions.TypeExtensions;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers;
+using osu.Framework.Threading;
 using osu.Game.Graphics;
 using osu.Game.Graphics.Containers;
 using osu.Game.Overlays;
@@ -47,9 +48,13 @@ protected override void LoadComplete()
             EditorBeatmap.TransactionEnded += updateInspectorText;
         }
 
+        private ScheduledDelegate? rollingTextUpdate;
+
         private void updateInspectorText()
         {
             inspectorText.Clear();
+            rollingTextUpdate?.Cancel();
+            rollingTextUpdate = null;
 
             switch (EditorBeatmap.SelectedHitObjects.Count)
             {
@@ -108,7 +113,7 @@ private void updateInspectorText()
 
                     // I'd hope there's a better way to do this, but I don't want to bind to each and every property above to watch for changes.
                     // This is a good middle-ground for the time being.
-                    Scheduler.AddDelayed(updateInspectorText, 250);
+                    rollingTextUpdate ??= Scheduler.AddDelayed(updateInspectorText, 250);
                     break;
 
                 default:
```

which - if my thinking is correct - will limit to at most one repeating delegate running at a time when single selection is active.

Reply:
Nice find. I originally wanted to use `AddOnce` because of this, but you can't delay that. Unfortunate we don't have a combination of both methods.